RA-04-create-and-show-fragment

DenisGubernatorov opened the pull request:
1) Поправлены префиксы значений в строковых ресурсах 2) В  gradle добавлена зависимость для использования фрагментов 3) Добавлен класс CategoriesListFragment наследующий Fragment и  файл лэйаут fragment_list_categories.xml 4) В основной лэйату добавлен контейнер для фрагментов 5) В MainActivity создается экземпляр фрагмента и запускается при создании активити

DenisGubernatorov committed:
RA-04-create-and-show-fragment
1) Поправлены префиксы значений в строковых ресурсах
2) В  gradle добавлена зависимость для использования фрагментов
3) Добавлен класс CategoriesListFragment наследующий Fragment и  файл лэйаут fragment_list_categories.xml
4) В основной лэйату добавлен контейнер для фрагментов
5) В MainActivity создается экземпляр фрагмента и запускается при создании активити

aGroupoid commented on app/src/main/java/com/example/recipesapp/CategoriesListFragment.kt:
```diff
@@ -0,0 +1,23 @@
+    ): View {
+        return binding.root
+    }
+}
```
Поскольку жизненный цикл фрагмента длиннее, чем у view внутри него, рекомендуется очищать binding в методе onDestroyView.
Подробнее можно почитать здесь: https://developer.android.com/topic/libraries/view-binding#fragments
Из-за этого внутри фрагментов придется отказаться от lazy. Поскольку очищать его не получится адекватно

aGroupoid commented on app/src/main/java/com/example/recipesapp/MainActivity.kt:
```diff
@@ -2,15 +2,21 @@ package com.example.recipesapp
         setContentView(binding.root)
+        val categoriesListFragment = CategoriesListFragment()
+        supportFragmentManager.commit {
+            add(R.id.mainContainer, categoriesListFragment)
```
Обращаю внимание на установку флага ​​​​setReorderingAllowed(true)​​​​. В [документации](https://developer.android.com/guide/fragments/transactions#reordering) этому посвящен отдельный абзац. Флаг нужен для корректного выполнения транзакции.

aGroupoid commented on app/src/main/java/com/example/recipesapp/CategoriesListFragment.kt:
```diff
@@ -0,0 +1,23 @@
+import androidx.fragment.app.Fragment
+import com.example.recipesapp.databinding.FragmentListCategoriesBinding
+
+class CategoriesListFragment : Fragment(R.layout.fragment_list_categories) {
```
Здесь передавать в конструктор не надо. Сначала создается макет через конструктор фрагмента, затем в onCreateView через binding.root. Лишний раз перерисовывать фрагмент - плохая практика

aGroupoid commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -8,15 +8,28 @@
         android:orientation="horizontal"
-        app:layout_constraintBottom_toBottomOf="parent">
+        app:layout_constraintBottom_toBottomOf="parent"
+        tools:layout_editor_absoluteX="16dp">
```
Нужно придерживаться, чтоб макет соответствовал всегда экрану реального приложеня. Этот параметр не нужен, он меняет изображение макета, хоть и не сильно, будет сложнее понять что у пользователей действительно на экране

DenisGubernatorov committed:
RA-04-create-and-show-fragment
1) Инициализация binding в CategoriesListFragment сделана через резервное свойство.
2) При выполнении транзакции в supportFragmentManager добавлен флаг  setReorderingAllowed
3) В LinearLayout убран лишний аргумент tools:layout_editor_absoluteX="16dp"

DenisGubernatorov commented:
Извини. Я помню, что ты просил не резолвить здесь комментарии. Я их зарезолвили в самой идешке. Ну и здесь изменения подтянулись. Что-то не подумал об этом. В следующий раз буду просто скрывать комментарии в идешке

aGroupoid reviewed (approved):
Всё верно, молодец! Резолвить не критично, но лучше так не делать. Попробуй покапаться в settings ИДЕшки, в разделе гитхаба, может там можно что-то изменить, чтоб локально резолвить но не на сайте
